src/components/gas/GasSpeedButton.js

```diff
@@ -0,0 +1,564 @@
+      .replace(nativeCurrencySymbol, '')
+      .trim();
+    return price;
+  }, [nativeCurrencySymbol, selectedGasFee]);
+
+  const isL2 = useMemo(() => isL2Network(currentNetwork), [currentNetwork]);
+
+  const formatGasPrice = useCallback(
+    animatedValue => {
+      // L2's are very cheap,
+      // so let's default to the last 2 significant decimals
+      if (isL2) {
+        const numAnimatedValue = Number.parseFloat(animatedValue);
+        if (numAnimatedValue < 0.01) {
+          return `${nativeCurrencySymbol}${numAnimatedValue.toPrecision(2)}`;
+        } else {
+          return `${nativeCurrencySymbol}${numAnimatedValue.toFixed(2)}`;
+        }
+      } else {
+        return `${nativeCurrencySymbol}${
+          nativeCurrency === 'ETH'
+            ? (Math.ceil(Number(animatedValue) * 10000) / 10000).toFixed(4)
+            : (Math.ceil(Number(animatedValue) * 100) / 100).toFixed(2)
+        }`;
+      }
+    },
+    [isL2, nativeCurrencySymbol, nativeCurrency]
+  );
+
+  const gasIsNotReady = useMemo(
+    () =>
+      isEmpty(gasFeeParamsBySpeed) ||
+      isEmpty(selectedGasFee?.gasFee) ||
+      isSufficientGas === null,
+    [gasFeeParamsBySpeed, selectedGasFee, isSufficientGas]
+  );
+
+  const openCustomGasSheet = useCallback(() => {
+    if (gasIsNotReady) return;
+    navigate(Routes.CUSTOM_GAS_SHEET, {
+      asset,
+      type: 'custom_gas',
+    });
+  }, [asset, navigate, gasIsNotReady]);
+
+  const openCustomOptions = useCallback(() => {
+    android && Keyboard.dismiss();
+    setShouldOpenCustomGasSheet(true);
+  }, [setShouldOpenCustomGasSheet]);
+
+  const renderGasPriceText = useCallback(
+    animatedNumber => (
+      <Text
+        color={
+          theme === 'dark'
+            ? colors.whiteLabel
+            : colors.alpha(colors.blueGreyDark, 0.8)
+        }
+        letterSpacing="roundedTight"
+        lineHeight="normal"
+        size="lmedium"
+        weight="bold"
+      >
+        {gasIsNotReady ? 'Loading...' : animatedNumber}
+      </Text>
+    ),
+    [theme, colors, gasIsNotReady]
+  );
+
+  const handlePressSpeedOption = useCallback(
+    selectedSpeed => {
+      if (selectedSpeed === CUSTOM && isEmpty(gasFeeParamsBySpeed[CUSTOM])) {
+        const gasFeeParams = gasFeeParamsBySpeed[selectedGasFeeOption];
+        updateToCustomGasFee({
+          ...gasFeeParams,
+          maxBaseFeePerGas: gasFeeParams.maxFeePerGas,
+          maxPriorityFeePerGas: gasFeeParams.maxPriorityFeePerGas,
+          option: CUSTOM,
+        });
+      } else {
+        updateGasFeeOption(selectedSpeed);
+      }
+    },
+    [
+      gasFeeParamsBySpeed,
+      selectedGasFeeOption,
+      updateToCustomGasFee,
+      updateGasFeeOption,
+    ]
+  );
+
+  const formatTransactionTime = useCallback(() => {
+    const time = parseFloat(estimatedTimeValue || 0).toFixed(0);
+    let selectedGasFeeGwei =
+      selectedGasFee?.gasFee?.estimatedFee?.value?.display;
```

Comment:
according to entities, it would be `display.display` ?

Reply:
removing it doesn't make sense to have that here since I changed the way we estimate times